[stable/prometheus-operator] Fix #15819 datasource config from k8s secrets

jnaulty opened the pull request:
Fix #15819 issue configuring datasources from kubernetes secrets

Changes in the stable/grafana charts allows for the recommended updating
and removal of datasources via kubernetes secrets. Older versions of
this chart do not allow this feature.

<!--
Thank you for contributing to helm/charts. Before you submit this PR we'd like to
make sure you are aware of our technical requirements and best practices:

* https://github.com/helm/charts/blob/master/CONTRIBUTING.md#technical-requirements
* https://github.com/helm/helm/tree/master/docs/chart_best_practices

For a quick overview across what we will look at reviewing your PR, please read
our review guidelines:

* https://github.com/helm/charts/blob/master/REVIEW_GUIDELINES.md

Following our best practices right from the start will accelerate the review process and
help get your PR merged quicker.

When updates to your PR are requested, please add new commits and do not squash the
history. This will make it easier to identify new changes. The PR will be squashed
anyways when it is merged. Thanks.

For fast feedback, please @-mention maintainers that are listed in the Chart.yaml file.

Please make sure you test your changes before you push them. Once pushed, a CircleCI
will run across your changes and do some initial checks and linting. These checks run
very quickly. Please check the results. We would like these checks to pass before we
even continue reviewing your changes.
-->

#### What this PR does / why we need it:
Addresses an issue/failure to configure datasources from kubernetes secrets

#### Which issue this PR fixes

  - fixes #15819 

#### Special notes for your reviewer:

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] [DCO](https://github.com/helm/charts/blob/master/CONTRIBUTING.md#sign-your-work) signed
- [x] Chart Version bumped
- [x] Variables are documented in the README.md
- [x] Title of the PR starts with chart name (e.g. `[stable/chart]`)

Commit:
[stable/prometheus-operator] Fix #15819

Fix #15819 issue configuring datasources from kubernetes secrets

Changes in the stable/grafana charts allows for the recommended updating
and removal of datasources via kubernetes secrets. Older versions of
this chart do not allow this feature.

Signed-off-by: John Naulty <johnnaulty@bitgo.com>

vsliouniaev reviewed (changes requested):
Since this is a feature, rather than a bug fix, please update the semver minor version

Commit:
[stable/prometheus-operator] bump up minor version

This is adding feature, so minor version should be updaated, not patch
version.

Signed-off-by: John Naulty <johnnaulty@bitgo.com>
